src/python/pants/base/exception_sink.py

```diff
@@ -0,0 +1,78 @@
+# coding=utf-8
+# Copyright 2018 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+
+from __future__ import absolute_import, division, print_function, unicode_literals
+
+import datetime
+import logging
+import os
+import sys
+from builtins import object
+
+from pants.util.dirutil import is_writable_dir, safe_open
+from pants.util.memo import memoized_classproperty
+
+
+logger = logging.getLogger(__name__)
+
+
+class ExceptionSink(object):
+  """A mutable singleton object representing where exceptions should be logged to."""
+
+  @memoized_classproperty
```

Comment:
A pattern we've used elsewhere to do the "mutable singleton" thing would be to ban calls to `__new__`, and then put all properties directly on the class.